### DISCUSSION TITLE
Use current user login (oaut2) to model

### DISCUSSION THREAD

**Discussion by Grazulex** [fastapi/fastapi#8910] [Questions]
Hello,
I'm new in FastApi and I try to used the user login into a model

My models are
```Python
from fastapi.datastructures import Default
from sqlalchemy import Column, Integer, String, ForeignKey, Boolean
from .database import Base
from sqlalchemy.orm import relationship

class User(Base):
    __tablename__ = 'users'

    id = Column(Integer, primary_key=True, index=True)
    firstname = Column(String(120))
    lastname = Column(String(120))
    email = Column(String(120))
    password = Column(String(120))
    hash = Column(String(120))
    is_valid = Column(Boolean, default=False)

    medias = relationship('Media', back_populates="owner")

class Media(Base):
    __tablename__ = 'medias'

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(120))
    body = Column(String(255))
    user_id = Column(Integer, ForeignKey('users.id'))

    owner = relationship("User", back_populates="medias")    
    
```

My schema are 
```Python
class ShowUser(BaseModel):
    firstname:str
    lastname:str
    email:str
    is_valid: bool
    hash: str
    medias : List[Media] =[]
    class Config():
        orm_mode = True   

class ShowMedia(BaseModel):
    title: str
    body:str
    owner: ShowUser

    class Config():
        orm_mode = True  
```

My repo is
```Python
from sqlalchemy.orm import Session
from .. import models, schemas
from fastapi import HTTPException,status, Depends

def create(request: schemas.Media,db: Session, current_user: schemas.SessionUser):
    new_media = models.Media(title=request.title, body=request.body, user_id=1)
    db.add(new_media)
    db.commit()
    db.refresh(new_media)
    return new_media
```

and my route is this
```Python
@router.post('/', status_code=status.HTTP_201_CREATED,)
def create(request: schemas.Media, db: Session = Depends(get_db),current_user: schemas.User = Depends(oauth2.get_current_user)):
    return media.create(request,db,current_user)
```

Issue is in the repo, this line :
```Python
new_media = models.Media(title=request.title, body=request.body, user_id=1
```

I want to change de user_id=1 to the current user login. I have used this in the route but when I used this variable in the route, this variable is empty (and I'm sure that the user is login)
```Python
current_user: schemas.User = Depends(oauth2.get_current_user)
```

How can i used the get_current_user in me repo ? thanks you

**Comment by sathoune** [comment]
Consider adding `id` field to your schemas. I suppose you did not include it because when you want to create the `User` you don't have `id` or have no way of knowing so. Check out this pattern in the template: https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/schemas/user.py#L25

There schemas differentiate between the function of the schema. There is one for creating user, updating and fetching, each conisdering different available fields. Usage of these models can be found here: https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/crud/crud_user.py#L15 Types are different between input and the output of the function.
